Makefile

```diff
@@ -157,6 +157,25 @@ run: go.build
 	@# To see other arguments that can be provided, run the command with --help instead
 	UPBOUND_CONTEXT="local" $(GO_OUT_DIR)/provider --debug
 
+# ====================================================================================
+# End to End Testing
+CROSSPLANE_NAMESPACE = upbound-system
+-include build/makelib/local.xpkg.mk
+-include build/makelib/controlplane.mk
+
+uptest: $(UPTEST) $(KUBECTL) $(KUTTL)
+	@$(INFO) running automated tests
+	@KUBECTL=$(KUBECTL) KUTTL=$(KUTTL) $(UPTEST) e2e "${UPTEST_EXAMPLE_LIST}" --setup-script=cluster/test/setup.sh || $(FAIL)
+	@$(OK) running automated tests
+
+local-deploy: build controlplane.up local.xpkg.deploy.provider.$(PROJECT_NAME)
+	@$(INFO) running locally built provider
+	@$(KUBECTL) wait provider.pkg $(PROJECT_NAME) --for condition=Healthy --timeout 5m
+	@$(KUBECTL) -n upbound-system wait --for=condition=Available deployment --all --timeout=5m
+	@$(OK) running locally built provider
+
+e2e: local-dev local-deploy
```

Comment:
what is `local-dev` target? Can't find it here or in the `build` submodule

Reply:
```
 make -n e2e
make: *** No rule to make target 'local-dev', needed by 'e2e'.  Stop.
```
make can't find it either, so looks like it will fail when invoked by the reusable workflow https://github.com/upbound/uptest/blob/main/.github/workflows/pr-comment-trigger.yml#L133

Reply:
Ah, great catch, it is just a mistake that I ended up while trying to find the best name for the `local-deploy` target 😅